adapters/MyTarget/MyTargetAdapter/GADMAdapterMyTargetNative.m

```diff
@@ -0,0 +1,264 @@
+	return [GADMAdapterMyTargetExtras class];
+}
+
+- (instancetype)initWithGADMAdNetworkConnector:(id<GADMAdNetworkConnector>)connector
+{
+	self = [super init];
+	if (self)
+	{
+		id<GADAdNetworkExtras> networkExtras = connector.networkExtras;
+		if (networkExtras && [networkExtras isKindOfClass:[GADMAdapterMyTargetExtras class]])
+		{
+			GADMAdapterMyTargetExtras *extras = (GADMAdapterMyTargetExtras *)networkExtras;
+			[GADMAdapterMyTargetUtils setLogEnabled:extras.isDebugMode];
+		}
+
+		MTRGLogInfo();
+		MTRGLogDebug(@"Credentials: %@", connector.credentials);
+		_connector = connector;
+	}
+	return self;
+}
+
+- (void)getBannerWithSize:(GADAdSize)adSize
+{
+	id<GADMAdNetworkConnector> strongConnector = _connector;
+	MTRGLogInfo();
+	guard(strongConnector) else return;
+	[strongConnector adapter:self didFailAd:[GADMAdapterMyTargetUtils errorWithDescription:kGADMAdapterMyTargetErrorBannersNotSupported]];
+}
+
+- (void)getInterstitial
+{
+	id<GADMAdNetworkConnector> strongConnector = _connector;
+	MTRGLogInfo();
+	guard(strongConnector) else return;
+	[strongConnector adapter:self didFailAd:[GADMAdapterMyTargetUtils errorWithDescription:kGADMAdapterMyTargetErrorInterstitialNotSupported]];
+}
+
+- (void)stopBeingDelegate
+{
+	MTRGLogInfo();
+	_connector = nil;
+	if (_nativeAd)
+	{
+		_nativeAd.delegate = nil;
+		_nativeAd = nil;
+	}
+}
+
+- (BOOL)isBannerAnimationOK:(GADMBannerAnimationType)animType
+{
+	return YES;
+}
+
+- (void)presentInterstitialFromRootViewController:(UIViewController *)rootViewController
+{
+	id<GADMAdNetworkConnector> strongConnector = _connector;
+	MTRGLogInfo();
+	guard(strongConnector) else return;
+	[strongConnector adapter:self didFailAd:[GADMAdapterMyTargetUtils errorWithDescription:kGADMAdapterMyTargetErrorInterstitialNotSupported]];
+}
+
+- (void)getNativeAdWithAdTypes:(NSArray *)adTypes options:(NSArray *)options
+{
+	id<GADMAdNetworkConnector> strongConnector = _connector;
+	MTRGLogInfo();
+	guard(strongConnector) else return;
+
+	NSUInteger slotId = [GADMAdapterMyTargetUtils slotIdFromCredentials:strongConnector.credentials];
+	guard(slotId > 0) else
+	{
+		MTRGLogError(kGADMAdapterMyTargetErrorSlotId);
+		[strongConnector adapter:self didFailAd:[GADMAdapterMyTargetUtils errorWithDescription:kGADMAdapterMyTargetErrorSlotId]];
+		return;
+	}
+	
+	_isContentAdRequested = [adTypes containsObject:kGADAdLoaderAdTypeNativeContent];
+	_isAppInstallAdRequested = [adTypes containsObject:kGADAdLoaderAdTypeNativeAppInstall];
+	_adTypesRequested = [adTypes componentsJoinedByString:@", "];
+
+	guard(_isContentAdRequested || _isAppInstallAdRequested) else
+	{
+		NSString *description = [NSString stringWithFormat:kGADMAdapterMyTargetErrorInvalidNativeAdType, _adTypesRequested];
+		MTRGLogError(description);
+		[strongConnector adapter:self didFailAd:[GADMAdapterMyTargetUtils errorWithDescription:description]];
+		return;
+	}
+
+	BOOL shouldDownloadImages = YES;
+	for (GADNativeAdImageAdLoaderOptions *imageOptions in options)
```

Comment:
The options are of type [`GADAdLoaderOptions`](https://developers.google.com/admob/ios/api/reference/Classes#/c:objc(cs)GADAdLoaderOptions). The sample code in the [Native Mediation Adapter](https://developers.google.com/admob/android/native-mediation) is incorrect. Please see the [SampleAdapter](https://github.com/googleads/googleads-mobile-ios-mediation/blob/8b819bf0063376920f0b7dc91487f2717894e8cd/example/Adapter/SampleAdapter.m#L155) for the correct implementation.